app/src/main/java/com/example/reduxtestapp/redux/middleware/PriceMiddleware.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.reduxtestapp.redux.middleware
+
+import com.example.reduxtestapp.data.repository.price.PriceRepository
+import com.example.reduxtestapp.redux.Action
+import com.example.reduxtestapp.redux.AppState
+import com.example.reduxtestapp.redux.Middleware
+import kotlinx.coroutines.delay
+import org.reduxkotlin.Store
+
+class PriceMiddleware(
+    private val repo: PriceRepository
+): Middleware() {
+    private var isPolling = false
+    override fun middleware(store: Store<AppState>, action: Any) {
+        when (action) {
+            is Action.Home.GetPrice -> {
+                store.dispatch(Action.Async{
+                    val result = repo.getPrice()
+                    store.dispatch(Action.Home.UpdatePrice(result))
+                })
+            }
+            is Action.Home.StartPolling -> {
+                println("[TEST] polling data")
+                isPolling = true
+                store.dispatch(Action.Async{
+                    store.dispatch(Action.Home.GetPrice)
+                    delay(action.intervalMillis)
+                    if (isPolling)
+                        store.dispatch(Action.Home.StartPolling(action.intervalMillis))
+                })
+            }
+            is Action.Home.StopPolling -> {
+                println("[TEST] stop")
+                isPolling = false
+            }
```

Comment:
left println statements so the start and stop of polling can be examined in logs